service/collector/env.go

```diff
@@ -41,19 +58,130 @@ func NewEnv(config EnvConfig) (*Env, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
-	v := &Env{
+	if config.Directory == "" {
+		config.Directory = "/run/flannel/networks/"
+	}
+
+	c := &Env{
 		g8sClient: config.G8sClient,
 		logger:    config.Logger,
+
+		directory: config.Directory,
 	}
 
-	return v, nil
+	return c, nil
 }
 
-func (v *Env) Collect(ch chan<- prometheus.Metric) error {
+func (c *Env) Collect(ch chan<- prometheus.Metric) error {
+	var desiredClusterIDs []string
+	{
+		l, err := c.g8sClient.Core().FlannelConfigs(metav1.NamespaceAll).List(metav1.ListOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, i := range l.Items {
+			desiredClusterIDs = append(desiredClusterIDs, i.Name)
+		}
+	}
+
+	var envClusterIDs []string
+	{
+		f := func(path string, info os.FileInfo, err error) error {
+			envClusterIDs = append(envClusterIDs, clusterIDFromPath(path))
```

Comment:
I understand there is possibility to have empty string in cluster ID. Is it a conscious decision?

Reply:
You mean because `clusterIDFromPath` returns an empty string when there is a file we cannot parse?

Reply:
Yes.

Reply:
Added error handling for empty strings.